docs/dev/architecture.md

```diff
@@ -1,25 +1,43 @@
 # Architecture
-The P4 Analyzer is to be packaged and deployed both as a native CLI executable, and a WebAssembly based Node.js package. The native executable will be built for Windows, Linux and MacOS, and configurable for use with any Language Server Protocol (LSP) client. The WebAssembly/Node.js build can be deployed intrisically as part of a Visual Studio Code extension, thereby negating the need to download any platform specific assets when the extension is first initialized.
+The P4 Analyzer is to be packaged and deployed both as a native CLI executable, and a WebAssembly based Node.js package.
+The native executable will be built for Windows, Linux and MacOS, and configurable for use with any Language Server
+Protocol (LSP) client. The WebAssembly/Node.js build will be deployed as part of a Visual Studio Code extension,
+thereby negating the need to download any platform specific assets when the extension is first initialized.
 
-This document describes some of the high level architectural decisions that have been made for the project, and can be used as a guide when looking at first navigating the codebase.
+This document describes some of the high level architectural pieces that exist in the project, and can be used as a
+guide when looking at navigating the codebase for the first time.
 
 ## <a name="analyzer-host"></a> The `AnalyzerHost` and `Analyzer`
-The `Analyzer` is responsible for ingesting P4 source code and producing a structured model of that P4 code that is highly optimized for querying. More specifically, P4 source code will be ingested by the `Analyzer` through a series of (\<_DocumentUri_\>, \<_Range_\>, `String`) tuples that represent both the initial contents, and then the subsequent changes or edits that are made to those source code files. The model maintained by the `Analyzer` is fully resolved and wholly maintained in memory, and as such, being derived from _change deltas_, will not be derived from any direct  I/O.
+The `Analyzer` is responsible for ingesting P4 source code and producing a structured model of that P4 code that is
+highly optimized for querying. The model maintained by the `Analyzer` is fully resolved and wholly maintained in
+memory, and as such, having no direct access to I/O, will be derived from the received LSP notifications directly.
+Request and Notification Handlers adapt the LSP messages to one or more calls to the core Analyzer. This design will
+allow the core Analyzer to be used outside of a Language Server context and in other tools such as Formatters and
+Linters.
 
-Since the `Analyzer` cannot perform any I/O directly, it will rely on a host to provide these (and other) services on its behalf. The following diagram shows this composition:
+Since the `Analyzer` cannot perform any I/O directly, it will rely on a host to provide these (and other) services
+on its behalf. The following diagram shows this composition:
```

Comment:
There seems to be a mixture of tenses used throughout this document. I suggest sticking with present simple ("it relies on a host to provide these services" rather than "it will rely on a host," "the `AnalyzerHost` filters responses" rather than "the `AnalyzerHost` will filter responses").